Wrong usage of CMAKE_MODULE_PATH in configure_package_config_file

I tried pulling in this project via [CPM](https://github.com/cpm-cmake/CPM.cmake) (a FetchContent CMake wrapper) like so
```
CPMAddPackage(
        NAME ale  # The unique name of the dependency (should be the exported target's name)
        GIT_TAG v0.9.0
        # Configuration options passed to the dependency (optional)
        SYSTEM TRUE
        BUILD_CPP_LIB=ON
        BUILD_PYTHON_LIB=OFF
        DOWNLOAD_ONLY FALSE       # If set, the project is downloaded, but not configured (optional)
        GITHUB_REPOSITORY Farama-Foundation/Arcade-Learning-Environment
)
```
but this leads to an error:
```
CMake Error at /u/michael.aichmueller/cmake-3.26.5-linux-x86_64/share/cmake-3.26/Modules/CMakePackageConfigHelpers.cmake:250 (message):
Unknown keywords given to CONFIGURE_PACKAGE_CONFIG_FILE():
“ale-config.cmake”
Call Stack (most recent call first):
cmake-build-debug-clang-lactose/_deps/ale-src/src/CMakeLists.txt:105 (configure_package_config_file)
```
due to using `CMAKE_MODULE_PATH` in `configure_package_config_file` as such
```
# Configure installable cmake config
  configure_package_config_file(
    ${CMAKE_MODULE_PATH}/${PROJECT_NAME}-config.cmake.in
    ${PROJECT_NAME}-config.cmake
    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
...
```
in `src/CMakeLists.txt`.
This leads to expanding the `CMAKE_MODULE_PATH` (which is a list) as multiple args blocking input AND output of the function, rendering `ale-config.cmake` to no longer be parsed as the output arg, but as a keyword arg (and consequently the error). Without knowing what the intentions were, this usage seems [incorrect.](https://discourse.cmake.org/t/configure-package-config-file-throws-error-of-unknown-keyword-project-name-config-cmake/10998/2?u=maichmueller)

This could be fixed by replacing `${CMAKE_MODULE_PATH}` with `${CMAKE_CURRENT_SOURCE_DIR}/../cmake` in the 1st argument.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -102,7 +102,7 @@ if (UNIX AND BUILD_CPP_LIB)
 
   # Configure installable cmake config
   configure_package_config_file(
-    ${CMAKE_MODULE_PATH}/${PROJECT_NAME}-config.cmake.in
+    ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/${PROJECT_NAME}-config.cmake.in
     ${PROJECT_NAME}-config.cmake
     INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
 

